.circleci/config.yml

```diff
@@ -0,0 +1,111 @@
+---
+version: 2.1
+
+executors:
+  ruby_executor:
+    working_directory: /mnt/ramdisk
+    docker:
+      - image: cimg/ruby:3.1.2-browsers
+  rails_executor:
+    working_directory: /mnt/ramdisk
+    docker:
+      - image: cimg/ruby:3.1.2-browsers
+        environment:
+          PGHOST: 127.0.0.1
+          PGUSER: root
+      - image: cimg/postgres:13.4
+        environment:
+          POSTGRES_USER: root
+          POSTGRES_DB: circle-test_test
+          POSTGRES_HOST_AUTH_METHOD: trust
+        command: postgres -c 'fsync=false' -c 'synchronous_commit=off'
+    resource_class: xlarge
+commands:
+  install_system_dependencies:
+    description: "Install system dependencies"
+    steps:
+      - run:
+          name: install system dependencies
+          command: |
+            # The CircleCI environment is based on Debian buster, which used to be called stable Debian, but is now called oldstable.
+            wget -q -O - https://dl.google.com/linux/linux_signing_key.pub | sudo apt-key add -
+            sudo apt-get update --allow-releaseinfo-change && \
+            sudo apt-get update && \
+            sudo apt-get install -y pdftk ghostscript
+  install_js_dependencies:
+    description: "Install JS dependencies"
+    steps:
+      - restore_cache:
+          key: yarn-{{ checksum "yarn.lock" }}
+      - run:
+          name: yarn install --frozen-lockfile
+          command: yarn install --frozen-lockfile
+      - save_cache:
+          key: yarn-{{ checksum "yarn.lock" }}
+          paths:
+            - /mnt/ramdisk/node_modules
+            - ~/.npm
+  install_ruby_dependencies:
+    description: "Install Ruby dependencies"
+    steps:
+      - restore_cache:
+          key: bundle-{{ checksum "Gemfile.lock" }}
+      - run:
+          name: install bundler & ruby dependencies
+          command: |
+            gem install bundler:2.3.5 --no-document && \
+            bundle install --jobs=4 --retry=3 --path vendor/bundle
+      - save_cache:
+          key: bundle-{{ checksum "Gemfile.lock" }}
```

Comment:
It seems `Gemfile.lock` is changing, which is resulting in the cache never being saved.

<img width="1624" alt="image" src="https://user-images.githubusercontent.com/67708639/202232671-06098723-2846-4706-807e-80c2636d5d27.png">

https://app.circleci.com/pipelines/github/codeforamerica/tax-benefits-foundational-project-2022-11/12/workflows/048513b8-1ee3-4129-941a-8dcd4150fc6b/jobs/15

Reply:
I think that if you use the `frozen` behavior of `bundle` you can detect and fix this problem. See https://stackoverflow.com/questions/68817050/using-bundle-frozen

Reply:
Added in f857d6a8e3d0a8d4a28eed65c56a4bbc0a2adc4b